packages/puppeteer-core/src/bidi/Page.ts

```diff
@@ -774,6 +775,136 @@ export class BidiPage extends Page {
     });
   }
 
+  override async cookies(...urls: string[]): Promise<Cookie[]> {
```

Comment:
I think it makes sense to use common/Cookie in all Page APIs related to cookies?

Reply:
`Cookie` and `CookieParams` are different. Do you suggest to copy CDP `CookieParams` as well?

Reply:
Right, I am not sure if we need to copy or if params could be derived from Cookie. Currently, the interface is like

```
abstract cookies(...urls: string[]): Promise<Cookie[]>;

abstract deleteCookie(
    ...cookies: Protocol.Network.DeleteCookiesRequest[]
  ): Promise<void>;
  
abstract setCookie(...cookies: Protocol.Network.CookieParam[]): Promise<void>;
```

I think it makes sense to remove CDP-specific structs from the API as part of the same breaking change (so that we do not need another breaking change).

Reply:
Perhaps `type DeleteCookiesRequest = Pick<Cookie, 'attr1'|'attr2'>` could be done like that? Similar with the CookieParam.

Reply:
It will end up with something like this, as some fields are optional, and `url` is not present in the `Cookie`. So I'm inclined to use a copy rather then this tricky `Pick`:
```
export type CookieParam =
  Pick<Cookie, 'name' |
    'value' |
    'sameSite' |
    'expires' |
    'priority' |
    'sameParty' |
    'sourceScheme' |
    'sourcePort' |
    'partitionKey'>
  & {
  url?: string;
  domain?: string;
  path?: string;
  secure?: boolean;
  httpOnly?: boolean;
};
```

Reply:
Copying sounds good to me.

Reply:
done